src/hotspot/share/prims/jvmtiExport.hpp

```diff
@@ -309,7 +309,7 @@ class JvmtiExport : public AllStatic {
   // If the jvmti_thread_state is absent and any thread filtered event
   // is enabled globally then it is created.
   // Otherwise, the thread->jvmti_thread_state() is returned.
-  static JvmtiThreadState* get_jvmti_thread_state(JavaThread *thread);
+  static JvmtiThreadState* get_jvmti_thread_state(JavaThread *thread, bool allow_suspend = true);
```

Comment:
Please add a detailed comment on what the `allow_suspend` parameter means and when it should be true and false.

Reply:
Good suggestion, thanks! Will add a comment.